packages/account-tree-controller/src/AccountTreeController.test.ts

```diff
@@ -4784,4 +4796,173 @@ describe('AccountTreeController', () => {
+        expect(groups[0].accounts).toContain(mockAccount1.id);
+        expect(groups[0].metadata.pinned).toBe(false);
+        expect(groups[0].metadata.hidden).toBe(true);
+        expect(mockCallbacks.isHiddenAccount).toHaveBeenCalledWith(
+          mockAccount1.id,
+        );
+      });
+
+      it('should prefer persisted metadata over callbacks', () => {
+        const mockIsHiddenAccount = jest.fn().mockReturnValue(true);
+        const mockIsPinnedAccount = jest.fn().mockReturnValue(true);
+
+        const { controller: tempController } = setup({
+          accounts: [mockAccount1],
+        });
+        tempController.init();
+        const tempWallets = Object.values(
+          tempController.state.accountTree.wallets,
+        );
+        const tempGroups = Object.values(tempWallets[0].groups);
+        const actualGroupId = tempGroups[0].id;
+
+        const { controller } = setup({
+          accounts: [mockAccount1],
+          state: {
+            accountGroupsMetadata: {
+              [actualGroupId]: {
+                pinned: {
+                  value: false,
+                  lastUpdatedAt: Date.now(),
+                },
+                hidden: {
+                  value: false,
+                  lastUpdatedAt: Date.now(),
+                },
+              },
+            },
+          },
+          config: {
+            backupAndSync: {
+              isAccountSyncingEnabled: true,
+              isBackupAndSyncEnabled: true,
+              onBackupAndSyncEvent: jest.fn(),
+            },
+            accountOrderCallbacks: {
+              isHiddenAccount: mockIsHiddenAccount,
+              isPinnedAccount: mockIsPinnedAccount,
+            },
+          },
+        });
+
+        controller.init();
+
+        // Verify callbacks were NOT called because persisted metadata takes precedence
+        expect(mockIsHiddenAccount).not.toHaveBeenCalled();
+        expect(mockIsPinnedAccount).not.toHaveBeenCalled();
+
+        const wallets = Object.values(controller.state.accountTree.wallets);
+        const groups = Object.values(wallets[0].groups);
+        expect(groups[0].accounts).toContain(mockAccount1.id);
+        expect(groups[0].metadata.pinned).toBe(false); // Persisted value used
+        expect(groups[0].metadata.hidden).toBe(false); // Persisted value used
+      });
+
+      it('should use persisted metadata when no callbacks are provided', () => {
+        const { controller: tempController } = setup({
+          accounts: [mockAccount1],
+        });
+        tempController.init();
+        const tempWallets = Object.values(
+          tempController.state.accountTree.wallets,
+        );
+        const tempGroups = Object.values(tempWallets[0].groups);
+        const actualGroupId = tempGroups[0].id;
+
+        const { controller } = setup({
+          accounts: [mockAccount1],
+          state: {
+            accountGroupsMetadata: {
+              [actualGroupId]: {
```

Comment:
That should work too:

```suggestion
        const walletId = toMultichainAccountWalletId(
          mockAccount1.options.entropy.id,
        );
        const groupId = toMultichainAccountGroupId(
          walletId,
          mockAccount1.options.entropy.groupIndex,
        );

        const { controller } = setup({
          accounts: [mockAccount1],
          state: {
            accountGroupsMetadata: {
              [groupId]: {
```